### PR TITLE
Update to allow for a Default to be set.

### PR DESCRIPTION
**Pull request by BinaryKitten**
Update the Taxonomy Single Term to allow a default to be specified, and if it's not found, fall back to the option for the default.
This is a better situation for new Posts where it hasn't been set and where the 1st item Alphabetically isn't the desired default.

### CODE REVIEW

File: class.taxonomy-single-term.php [hunk 1]
```diff
@@ -112,18 +112,28 @@ class Taxonomy_Single_Term {
 	 * @var boolean
 	 */
 	protected $allow_new_terms = false;
+	
+	/**
+	 * Default for the selector
+	 * @since 0.2.2
+	 * @var array
+	 */
+	protected $default = [];
```
Comment (jtsternberg) [inline class.taxonomy-single-term.php]: this lib supports same version of php as WordPress, so shorthand notation won't work.

File: class.taxonomy-single-term.php [hunk 2]
```diff
@@ -134,6 +144,37 @@ public function __construct( $tax_slug, $post_types = array(), $type = 'radio' )
...
+	 * @param array $default
+	 * @param $tax_slug
+	 *
+	 * @return array
+	 */
+	protected function process_default( $default = array(), $tax_slug ) {
+		$default = (array) $default;
+
+		if ( empty( $default ) ) {
+			$default = [ (int) get_option( 'default_' . $tax_slug ) ];
```
Comment (jtsternberg) [inline class.taxonomy-single-term.php]: More shorthand.

**Comment by BinaryKitten** [comment]
Hi @jtsternberg 

I've needed the changes on a project, so have tested in that environment. (WordPress 4.7.5 Woocommerce on Product Post type)

**Comment by jtsternberg** [comment]
One other request. The `process_default` method does not need to take a `$tax_slug` param, since in both cases it simply uses `$this->slug` anyway. That will help simply that method.

**Comment by BinaryKitten** [comment]
Removed. at the time it made sense, but as you say it's part of the class anyway.

**Comment by jtsternberg** [comment]
Thank you!

**Pull request merged**